### DISCUSSION TITLE
Remove need to click "when was this page last modified?" button

### DISCUSSION THREAD

**Discussion by rgov** [gollum/gollum#1852] [General]
The footer of every page has a button, "When was this page last modified?"

This button trigger some JavaScript ([implemented here](https://github.com/gollum/gollum/blob/6f870501a0d4b038f355382269f61915f1f6484b/lib/gollum/public/gollum/javascript/gollum.js.erb#L589-L602)), which makes a request to the API at `/gollum/last_commit_info?path=Home.md` and then substitutes it into the page.

Is there a compelling reason why this information could not be served simultaneously with the page content? Or at least the API call made automatically when the page is loaded?

**Comment by dometto** [comment]
Basically it's a performance issue. See the PR that implemented the ajax call for info: https://github.com/gollum/gollum/pull/1155

It's possible that `rugged` has implemented the relevant logic in a faster way now (using C rather than Ruby) -- I haven't checked. If you feel strongly about the issue and wanted to investigate that, such help would be very welcome. I'll turn this into a Discussion for now.
